apidef/oas/default.go

```diff
@@ -126,6 +144,44 @@ func (as *AuthSources) Import(in string) {
 	}
 }
 
+func (s *OAS) importMiddlewares(allowList, validateRequest *bool) {
+	xTykAPIGateway := s.GetTykExtension()
+	middleware := xTykAPIGateway.Middleware
+
+	if middleware == nil {
+		middleware = &Middleware{}
+		xTykAPIGateway.Middleware = &Middleware{}
+	}
+
+	for path, pathItem := range s.Paths {
+		for _, method := range allowedMethods {
+			if operation := pathItem.GetOperation(method); operation != nil {
+				tykOperation := s.getTykOperation(method, path)
+				tykOperation.Import(operation, allowList, validateRequest)
+				s.deleteTykOperationIfEmpty(tykOperation, method, path)
```

Comment:
there are cases when tykOperation is having zero values (got from `s.getTykOperation`). it leads to Operations to have empty entry in the map. `deleteTykOperationIfEmpty` will delete such entries